### ISSUE TITLE
metadata.extension is singular - should be an array instead?

### ISSUE THREAD

**Issue by mikeradka** [ocsf/ocsf-schema#808]
To log an extension associated with an OCSF event, we have the [extension object](https://schema.ocsf.io/1.0.0/objects/extension?extensions=) (eg `metadata.extension`)

<img width="681" alt="image" src="https://github.com/ocsf/ocsf-schema/assets/91983279/e5bcb47d-b515-489a-bc79-bdd78fa159e7">

That said, there are use-cases for adopters to use multiple extensions for OCSF events.

Is there benefit in adding an `extensions` object, composed of an array of `extension`?

Additionally - I can see why `metadata.extension` would not be required for the base event, as different adopters have different storage concerns. But from an auditing and data integrity standpoint, it seems a bit more important than `Optional`. Should we bump `metadata.extension` to `Recommended`?

**Comment by mikeradka** [comment]
One common use-case for multiple extensions:

- The ACME organization has their own internal OCSF extension
- Because they are a Windows shop, they also use the Windows extension in combination with the internal ACME extension

Also a point to raise: simply moving from a single string to an array of strings could potentially be breaking. It might be better instead to consider simply adding the plural `metadata.extensions` and deprecating `metadata.extension`

**Comment by pagbabian-splunk** [comment]
The case where a Platform extension and a vendor extension would apply to a single event is the driving factor here IMO.
It would be breaking if we replaced `extension` with `extensions`: the pattern we have used is to deprecate `extension` as you suggest.

Re: Optional.  Since extensions themselves are optional, I don't see why we would make `extension` or `extensions` Recommended.  That is, we aren't recommending that people always write an extension.
